### DISCUSSION TITLE
Loading Pre-Defined Templates using multi option picker

### DISCUSSION THREAD

**Discussion by sghriyadh** [Budibase/budibase#10578] [Help] [answered]
I have 3 users and each one has his own template to write Medical report.

user1 has 2 templates
user2 has 4 templates
user3 has 3 templates

How can i load a pre defined template using a option picker?

i am a no coder

can anyone share a small app which i can modify for my use?

**Comment by melohagan** [comment]
Hey @sghriyadh 

I don't really have enough information to explore your use-case, however this docs page may give you inspiration! https://docs.budibase.com/docs/filter-by-relationships

**Reply by sghriyadh** [reply]
May be i mis-wrote my question.

i will explain this with another example.

I want to create a form about Mobiles types and their specifications.

Mobile types will be in option picker and specifications will be a long text field.

If user selects Samsung 720 Model the specifications load the long text field with details of that model (default details for that model). I want user to edit these details and update them.

Any already made  template that i can use to learn who this works in Budibase as i am not a coder by profession but i can customize the already made templates.

**Reply by ConorWebb96** [reply]
Hey @sghriyadh, 

What sort of database are you using? Is it the regular Budibase DB or an external such as: MySQL, rest etc?

Thanks in advance for the additional information.

**Comment by sghriyadh** [comment]
Regular Budibase DB

**Reply by ConorWebb96** [reply, marked as answer]
Hey @sghriyadh,

I believe i have achieved the functionality you mentioned earlier. Below are some screenshots of tables and components as well as their setup.

**Table setup**
<img width="1914" alt="Screenshot 2023-05-22 at 13 25 25" src="https://github.com/Budibase/budibase/assets/126772285/24ae177d-4fd7-4c33-ada4-59a069530b71">

**Screen with components**
<img width="1872" alt="Screenshot 2023-05-22 at 13 25 45" src="https://github.com/Budibase/budibase/assets/126772285/5f68cca6-1f89-4da4-8894-3d048805b80d">

**Options picker settings**
<img width="301" alt="Screenshot 2023-05-22 at 13 25 54" src="https://github.com/Budibase/budibase/assets/126772285/39141032-8550-40a1-b824-f25a79aa96a7">

**Fetch row action on change of options picker**
<img width="1299" alt="Screenshot 2023-05-22 at 13 26 01" src="https://github.com/Budibase/budibase/assets/126772285/b455e7cb-1eed-454f-91db-326ffebd4b62">

**Update field action using the fetch row data**
<img width="1298" alt="Screenshot 2023-05-22 at 13 26 08" src="https://github.com/Budibase/budibase/assets/126772285/d1c17e42-ee3f-4228-8fd2-e9391b0e7062">

**Demo of the result**
![Loom Message - 22 May 2023](https://github.com/Budibase/budibase/assets/126772285/eae8e4d7-3410-48fe-a8a9-33dc0d45873a)

Hopefully this helps to point you in the right direction.

**Comment by sghriyadh** [comment]
It's a brilliant effort, as i see it is loading various specifications values based on Title value selected from option picker.

How can i add Name of the mobile owner to the same page and save it as a "New record". Like creating a directory of different mobile owners with their names, mobile number using this option picker integration that you have created?

**Reply by ConorWebb96** [reply]
Hey @sghriyadh,

1. First bind up the form to where you want the submitted data to go for example; `User Selection`.
2. Add a button to the form supply it with various actions; `Validate form`, `Save Row` and `Clear form`.
3. Add in different fields, in the above example specification is already there, so the only additional fields you'll need are; title, name and phone number.
4. From here you can add an additional `Update field value` action for the select field. Then do something similar to what was shown above for the specification field, the update state will likely look something like this; `{{ Action 1.Fetched row.Title }}`.
5. Setup validation rules if needed can read through this: https://docs.budibase.com/docs/form-validation should help with this.
6. If done correctly you will have fields pre populated with data related to the selection and will only need to fill out the name and number field before saving.

Hopefully this all makes sense.

**Comment by sghriyadh** [comment]
I am not lazy but this setup went over my head.

Can you create a small basic app and share me link for it's zip. it's easy for me to customize than to create . Sorry to ask you for alot but a zipped app will be a great help to me. 

Regards

**Comment by sghriyadh** [comment]
@ConorWebb96 

Here is the app i created  as zip file.if you can review 

https://www.sendgb.com/upload/?utm_source=6JkaMgmosIW

As you can see in it i want to "select template" and want to load it into the box "loaded template"

If you or anyone can fix this for me?  Thank you in advance.

**Comment by ConorWebb96** [comment]
Hey @sghriyadh,

Apologies I actually did make a example for you, i forgot to send it across yesterday evening. You can download the app export below. 
[sghriyadh example export.tar.gz](https://github.com/Budibase/budibase/files/11553890/sghriyadh.example.export.tar.gz)

Specification within this example is the screen you are after. Home and types are just to generate/see the data.

Hopefully this aids in some way. If not I will look into your app export you provided.

**Comment by sghriyadh** [comment]
Thank you. 

i downloaded your app and i can customize and add more fields to it as was my requirement.

Sorry but any template of this type wasn't found in template library so i had to ask for help.

Regards

**Comment by sghriyadh** [comment]
@ConorWebb96 

i customized your template and it's great. only one thing i cannot figure out.

Once a user click the modified specifications, it opens side panel. i want to open it on a separate full page(screen) (not modal or side panel). 

Can you modify your template for that and repost the link of new zip again? 

i hope that it's not a big ask. Thank you.

**Reply by melohagan** [reply]
I appreciate it's a late response, but for anyone who may happen to stumble upon it there is a docs page with more information on this https://docs.budibase.com/docs/url-parameters
